detect/detect.go

```diff
@@ -56,6 +56,10 @@ type Detector struct {
 	// IgnoreGitleaksAllow is a flag to ignore gitleaks:allow comments.
 	IgnoreGitleaksAllow bool
 
+	// TODO: Does it make sense for |RepositoryUrl| to be a property on Detector?
+	// If someone is using Gitleaks as a library, could they reuse a detector across multiple repositories?
+	RepositoryUrl string
```

Comment:
Thoughts @bplaxco @zricethezav? I decided to pass `scm.Provider` as an argument to `DetectGit` for this reason.

Reply:
@rgmz I generally create a new detector per repo since things like findings and commitMap exist on the detector and at an earlier time things like keywords and such where on the detector. 

Tweaking DetectGit would cause a small change in my stuff but it wouldn't be a big one. 

I wonder if it would make sense to include these things on GitCmd. Then you'd have all the context for the repo in one place (the provider, the URL, the content from the repo, etc).

Reply:
> I generally create a new detector per repo since things like findings and commitMap exist on the detector and at an earlier time things like keywords and such where on the detector.

Hmm, you're right. Perhaps removing state from Detector could be done in the future.

Passing a struct to `DetectGit` is a great idea, and would prevent further cluttering of `Detector`.